libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Views/SFSDKWindowContainer.h

```diff
@@ -0,0 +1,176 @@
+/*
+ SFSDKWindowContainer.h
+ SalesforceSDKCore
+ 
+ Created by Raj Rao on 7/4/17.
+ 
+ Copyright (c) 2017-present, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
+typedef NS_OPTIONS(NSUInteger, SFSDKWindowType) {
+    SFSDKWindowTypeMain        = 0,
+    SFSDKWindowTypeAuth        = 1 << 0,
+    SFSDKWindowTypePasscode    = 1 << 1,
+    SFSDKWindowTypeSnapshot    = 1 << 2,
+    SFSDKWindowTypeOther       = 1 << 3
+};
+@class SFSDKWindowContainer;
+
+@protocol SFSDKWindowContainerDelegate<NSObject>
+
+@optional
+
+/**
+ Called when the window is going to be made key & visible
+ @param window The window that is going to be made key & visible
+ */
+- (void)windowWillMakeKeyVisible:(SFSDKWindowContainer *)window;
```

Comment:
New classes should define nullability constraints on all methods and properties.